tcp/spring-cloud-starter-stream-source-tcp-client/src/main/java/org/springframework/cloud/stream/app/tcp/client/source/TcpClientSourceConfiguration.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.app.tcp.client.source;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.stream.annotation.EnableBinding;
+import org.springframework.cloud.stream.app.tcp.EncoderDecoderFactoryBean;
+import org.springframework.cloud.stream.app.tcp.client.source.TcpClientSourceProperties;
+import org.springframework.cloud.stream.messaging.Processor;
+import org.springframework.cloud.stream.messaging.Source;
+import org.springframework.context.annotation.Bean;
+import org.springframework.integration.annotation.ServiceActivator;
+import org.springframework.integration.ip.config.TcpConnectionFactoryFactoryBean;
+import org.springframework.integration.ip.tcp.TcpReceivingChannelAdapter;
+import org.springframework.integration.ip.tcp.TcpSendingMessageHandler;
+import org.springframework.integration.ip.tcp.connection.AbstractConnectionFactory;
+import org.springframework.integration.ip.tcp.connection.TcpMessageMapper;
+import org.springframework.integration.ip.tcp.serializer.AbstractByteArraySerializer;
+
+/**
+ * A source application that acts as a TCP client.
+ *
+ * @author Ilayaperumal Gopinathan
+ * @author Gary Russell
+ */
+@EnableBinding(Source.class)
+@EnableConfigurationProperties(TcpClientSourceProperties.class)
+public class TcpClientSourceConfiguration {
+
+	@Autowired
+	private Source source;
+
+	@Autowired
+	private TcpClientSourceProperties properties;
+
+	@Bean
+	public TcpReceivingChannelAdapter adapter(
+			@Qualifier("tcpClientConnectionFactory") AbstractConnectionFactory connectionFactory) {
+		TcpReceivingChannelAdapter adapter = new TcpReceivingChannelAdapter();
+		adapter.setConnectionFactory(connectionFactory);
+		adapter.setClientMode(true);
```

Comment:
When using client mode, we should also expose the `retryInterval` as a property - it controls how often we check the connection and reconnect if necessary - default is 60000.